docs/authentication.md

```diff
@@ -0,0 +1,134 @@
+            * Used for signing a request.
+    * Ledger-side properties
+      * `scalar.dl.ledger.authentication.method` (set "digital-signature")
+    * Auditor-side properties
+      * `scalar.dl.auditor.authentication.method` (set "digital-signature")
+* Ledger authentication 
+    * Ledger-side properties
+        * `scalar.dl.ledger.proof.enabled` (set true)
+            * Required because Ledger authentication uses the signatures of asset proofs.
+        * `scalar.dl.ledger.proof.private_key_pem` or `scalar.dl.ledger.proof.private_key_path`  
+            * Used for signing asset proofs.
+    * Auditor-side properties
+        * `scalar.dl.auditor.ledger.cert_holder_id`
+        * `scalar.dl.auditor.ledger.cert_version`
+            * Used for verifying the signatures of asset proofs.
+* Auditor authentication
+    * Ledger-side properties
+        * `scalar.dl.ledger.auditor.cert_holder_id`
+          * Used for verifying an Auditor-generated signature attached to a client request.
+        * `scalar.dl.ledger.auditor.cert_version`
+          * Used for verifying an Auditor-generated signature attached to a client request.
+    * Auditor-side properties
+        * `scalar.dl.auditor.cert_holder_id`
+          * Used for calling Ledger services directly by using the client library.
+        * `scalar.dl.auditor.cert_version`
+          * Used for calling Ledger services directly by using the client library.
+        * `scalar.dl.auditor.private_key_pem` or `scalar.dl.auditor.private_key_path`
+          * Used for signing a client request and a request from Auditor to Ledger.
+
+### HMAC
+
+* Client authentication
+    * Client-side properties
+        * `scalar.dl.client.authentication_method` (set "hmac")
+        * `scalar.dl.client.entity.id`
+            * Used for identifying a client.
+        * `scalar.dl.client.entity.identity.hmac.secret_key`
+            * Used for signing a request.
+        * `scalar.dl.client.entity.identity.hmac.key_version`
+    * Ledger-side properties
+      * `scalar.dl.ledger.authentication.method` (set "hmac")
+    * Auditor-side properties
+      * `scalar.dl.auditor.authentication.method` (set "hmac")
+
+* Ledger and Auditor authentication
+    * Ledger-side properties
+        * `scalar.dl.ledger.proof.enabled` (set true)
+            * Required because Ledger authentication uses the signatures of asset proofs.
+        * `scalar.dl.ledger.authentication.hmac.cipher_key`
+            * Used for signing and verifying messages and requests between Ledger and Auditor.
+    * Auditor-side properties
+        * `scalar.dl.auditor.authentication.hmac.cipher_key`
+            * Used for signing and verifying messages and requests between Ledger and Auditor. Must be the same key as `scalar.dl.ledger.authentication.hmac.cipher_key`.
```

Comment:
According to the comment of [ledger/conf/ledger.properties](https://github.com/scalar-labs/scalar/blob/3.8/ledger/conf/ledger.properties), `scalar.dl.ledger.authentication.hmac.cipher_key` is explained as following:
```java
# A cipher key used to encrypt and decrypt the HMAC secret keys of client entities.
...
```
and `scalar.dl.ledger.servers.authentication.hmac.secret_key` is explained as following:
```java
# A secret key of HMAC for the authentication of messages between (Ledger and Auditor) servers. The same key has to be set in the corresponding Auditor as well.
...
```
So, Maybe the description of `scalar.dl.ledger.authentication.hmac.cipher_key` is not correct. (I don't have strong confidence. maybe..)?
It may explain for `scalar.dl.ledger.servers.authentication.hmac.secret_key` not for `scalar.dl.ledger.authentication.hmac.cipher_key`?
(Sorry If I misunderstood.)